paimon-web-common/src/main/java/org/apache/paimon/web/common/executor/Executor.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.common.executor;
+
+import org.apache.paimon.web.common.result.FetchResultParams;
+import org.apache.paimon.web.common.result.SubmitResult;
+
+/** The Executor interface. */
+public interface Executor {
+
+    SubmitResult executeSql(String statement) throws Exception;
+
+    SubmitResult fetchResults(FetchResultParams params) throws Exception;
+
+    boolean stop(String jobId, boolean withSavepoint, boolean withDrain) throws Exception;
```

Comment:
Is the executor only defined for Flink job? What's the behavior of Spark job for `withSavepoint` and `withDrain` param?

Reply:
Spark's stop method may not need withSavepoint and withDrain, I will pass a variable parameter.